src/cargo/core/resolver/types.rs

```diff
@@ -66,7 +67,17 @@ impl ResolverProgress {
         // If any of them are removed then it takes more than I am willing to measure.
         // So lets fail the test fast if we have ben running for two long.
         if cfg!(debug_assertions) && (self.ticks % 1000 == 0) {
-            assert!(self.start.elapsed() - self.deps_time < Duration::from_secs(90));
+            // Some CI setups are much slower then the equipment used by Cargo itself.
+            // Architectures that do not have a modern processor, hardware emulation, ect.
+            // In the test code we have `slow_cpu_multiplier`, but that is not accessible here.
+            lazy_static::lazy_static! {
```

Comment:
Instead of `lazy_static`, could this be stored as a field in `ResolverProgress`?

Reply:
Does that look better?